test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/IdentityHashMapTest.scala

```diff
@@ -0,0 +1,397 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.util
+
+import java.{util => ju}
+import scala.reflect.ClassTag
+import org.junit.Assert._
+import org.junit.Test
+
+class IdentityHashMapTest extends MapTest {
+  import MapTest._
+
+  override def factory: IdentityHashMapFactory = new IdentityHashMapFactory
+
+  private val testObjCache = new ju.HashMap[Int, TestObj]()
+
+  override def testObj(i: Int): TestObj = {
+    val existing = testObjCache.get(i)
+    if (existing ne null) {
+      existing
+    } else {
+      val newTestObj = TestObj(i)
+      testObjCache.put(i, newTestObj)
+      newTestObj
+    }
+  }
+
+  def assumeFalseIdentityHashMapViews(): Unit = assumeFalseIdentityHashMap("JVM views removeAll, retainAll")
+
+  // Needs false negative tests
+  // Tests that check by value rather than identity
+  // Similar to tests already in MapTest
```

Comment:
Could you also add false negative tests for the basic methods: `containsKey`, `get`, `containsValue`, `remove`.

And also a test where you `put` two keys that are `equals` but not `eq` in the map, and verify that you can `get` their respective associated values, `containsKey` them and `remove` them.

Reply:
I added an additional test case and more tests in each of the respective `remove` sections with no surprises.